tests/wasmfs/wasmfs_before_preload.c

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2022 The Emscripten Authors.  All rights reserved.
+ * Emscripten is available under two separate licenses, the MIT license and the
+ * University of Illinois/NCSA Open Source License.  Both these licenses can be
+ * found in the LICENSE file.
+ */
+
+#include <assert.h>
+#include <emscripten/console.h>
+#include <emscripten/wasmfs.h>
+#include <stdio.h>
+
+static backend_t my_js_file_backend;
```

Comment:
Unrelated to this PR but shouldn't `backend_t` be in a namespace?  I don't see any `using` directives in this file?

Reply:
No, because it's part of the C (not C++) wasmfs API.

Reply:
Yes.. C apis should probably use name mangling like that in place of namespaces.